src/libraries/System.Collections/tests/Generic/Dictionary/Dictionary.Tests.cs

```diff
@@ -337,6 +337,22 @@ public void CantAcceptDuplicateKeysFromSourceDictionary()
             AssertExtensions.Throws<ArgumentException>(null, () => new Dictionary<string, int>(source, StringComparer.OrdinalIgnoreCase));
         }
 
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotInvariantGlobalization))]
+        // https://github.com/dotnet/runtime/issues/44681
+        public void DictionaryOrdinalIgnoreCaseCyrillicKeys()
+        {
+            const string Lower = "абвгдеёжзийклмнопрстуфхцчшщьыъэюя";
+            const string Higher = "АБВГДЕЁЖЗИЙКЛМНОПРСТУФХЦЧШЩЬЫЪЭЮЯ";
```

Comment:
Nit, coding-style.md

> When including non-ASCII characters in the source code use Unicode escape sequences (\uXXXX) instead of literal characters. Literal non-ASCII characters occasionally get garbled by a tool or editor.

No need to fix it in this PR though.

Reply:
I fully appreciate where this guideline came from, but the trouble with it - it makes the code significantly harder to read and maintain (pertaining to this specific example).

Reply:
Yeah. IMO when our unit tests use strings in non-English languages, we should just write the literal strings unescaped in their original language. If our unit tests are instead testing "I put a non-ASCII character here, let's see what happens!", then it's good to put the "\uXXXX" explicitly, since it draws attention to the fact that there's something unique about the character at this index in the string.

Reply:
I agree it's harder to read -- the concern is if garbelizing could somehow allow the test to continue to pass, without anyone notice.

Another option is to follow it with a Debug.Assert that compares them to their escaped forms. That way you maintain the readability, but garbelizing will immediately fail the test.

I don't feel strongly - if you think it is impossible that we would not notice.